server/route.go

```diff
@@ -1037,17 +1037,16 @@ func (c *client) processRemoteSub(argo []byte, hasOrigin bool) (err error) {
 	if acc == nil {
 		expire := false
 		isNew := false
-		if !srv.NewAccountsAllowed() {
-			// if the option of retrieving accounts later exists, create an expired one.
-			// When a client comes along, expiration will prevent it from being used,
-			// cause a fetch and update the account to what is should be.
-			if staticResolver {
-				c.Errorf("Unknown account %q for remote subject %q", accountName, sub.subject)
-				return
-			}
-			c.Debugf("Unknown account %q for remote subject %q", accountName, sub.subject)
-			expire = true
+		// if the option of retrieving accounts later exists, create an expired one.
+		// When a client comes along, expiration will prevent it from being used,
+		// cause a fetch and update the account to what is should be.
+		if staticResolver {
+			c.Errorf("Unknown account %q for remote subject %q", accountName, sub.subject)
+			return
 		}
+		c.Debugf("Unknown account %q for remote subject %q", accountName, sub.subject)
+		expire = true
```

Comment:
No big deal, but expire is initialized to false but now unconditionally set to true, so we could remove `expire` all together (including in the test just below)

Reply:
Will do.

Reply:
Fixed.